pkg/gadgets/run/types/types.go

```diff
@@ -19,15 +19,34 @@ import (
 	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
 )
 
+type Endpoint struct {
+	eventtypes.L4Endpoint
+	Version uint8  `json:"version,omitempty"`
+	Name    string `json:"name,omitempty"`
+}
+
 type Event struct {
 	eventtypes.Event
 	eventtypes.WithMountNsID
+
+	Endpoints []Endpoint `json:"endpoints,omitempty"`
+
 	// Raw event sent by the ebpf program
-	RawData []byte `json:"raw_data"`
+	RawData []byte `json:"raw_data,omitempty"`
 	// How to flatten this?
 	Data interface{} `json:"data"`
 }
 
+func (ev *Event) GetEndpoints() []*eventtypes.L3Endpoint {
```

Comment:
Good point. We have the same problem for operators checking `SetMountNsMap`. I think we would need to change the initialization logic to have a sort of "PreRun" to allow the gadget to configure itself, then the operators should not check the gadget desc but the tracer instance to see if they can operate on that. (Not only based on the implementation of an interface but on the return value of a method).  @flyth any thoughts on this?

Reply:
I totally agree with the first statement: users should be able to disable expensive operators. On the second one, I think `CanOperateOn` still needs some knowledge about the gadget: even if the operator is enabled, we should avoid creating and instance of it if it's not compatible with a given gadget.

Reply:
An operators' Instantiate() can return nil in case it doesn't want to be enabled (at least that's how it was planned). In Instantiate() you should have all the necessary information (e.g. operator params), right? CanOperateOn should be just used for the compatibility check.

Reply:
I did some experiments in https://github.com/inspektor-gadget/inspektor-gadget/commit/4a4f3db719b68a98f01efa18f701de10cacca972. It doesn't look like a perfect solution to me but something in that direction could work fine. My main concern is about the overlap with `CanOperate()`, should we keep both of them?

Reply:
The issue is that they work at different levels. `CanOperateOn` is more static: it only uses interfaces and data types to understand if a given operator can work on a gadget. It made sense for "built-in" gadgets, but the `run` gadget is do dynamic that depending on the actual gadget being run some operator make sense and others don't. 

I'd say that we can keep both of them right now, once we start deprecating (and removing) built-in gadgets the need for  `CanOperateOn`  could disappear.